### ISSUE TITLE
ORPO preference scoring includes system and user prompt tokens instead of response tokens only

### ISSUE THREAD

**Issue by Sayitobar** [Goekdeniz-Guelmez/mlx-lm-lora#79]
## Summary

In `mlx_lm_lora 3.1.2`, the ORPO batching and scoring path creates masks that include every non-padding token in both the chosen and rejected sequences.

As a result, `get_logps()` includes system and user prompt tokens in the preference score, even though those tokens are conditioning context rather than part of the chosen or rejected response.

The relevant functions are:

- `mlx_lm_lora.trainer.orpo_trainer.iterate_orpo_batches()`
- `mlx_lm_lora.trainer.orpo_trainer.get_logps()`

## Environment

- `mlx_lm_lora`: 3.1.2
- `mlx`: 0.32.0
- `mlx-lm`: 0.31.3
- Python: 3.10.18
- OS: macOS 15.7.3 on Apple Silicon (M1 Pro)
- `transformers`: 5.12.1

The same behavior was also present in `mlx_lm_lora 2.1.0`.

## Minimal reproduction

A constructed batch-level check with the following non-padding token sequences:

```text
chosen:   [10, 11, 12]
rejected: [10, 13]
```

produced the following masks, padded to length 8:

```text
chosen mask:   [1, 1, 1, 0, 0, 0, 0, 0]
rejected mask: [1, 1, 0, 0, 0, 0, 0, 0]
```

The batcher does not preserve a response-start boundary. Therefore, if the first two token positions represent the shared prompt and the final non-padding position represents the response, a response-only implementation would need masks such as:

```text
chosen:   [0, 0, 1, 0, 0, ...]
rejected: [0, 0, 1, 0, 0, ...]
```

No model forward pass or training run is required to observe the mask construction.

## Actual behavior

`iterate_orpo_batches()` creates masks over the full non-padding sequence. `get_logps()` then applies those masks while averaging token log probabilities.

The scorer computes logits from:

```python
tokens[:, :-1]
```

and targets from:

```python
tokens[:, 1:]
```

but applies the mask as:

```python
mask[:, :-1]
```

The latter is a separate next-token mask alignment issue. Even ignoring that separate issue, the mask still includes prompt positions.

Operationally, the current path scores the rendered sequence rather than only the assistant response:

```text
P(system + user prompt + response)
```

rather than:

```text
P(response | system + user prompt)
```

## Expected behavior

The ORPO scoring mask should be:

- `0` over system and user conditioning tokens;
- `1` over assistant-response target tokens;
- `0` over padding.

## Why this matters

Prompt tokens are conditioning context, not preference targets. Including them in the average:

- dilutes the chosen-versus-rejected response signal;
- makes the score depend on prompt length and chat-template overhead;
- includes tokens that are often identical between chosen and rejected sequences;
- can make preference metrics reflect prompt likelihood more than response quality.

This issue is independent of the separate `mask[:, :-1]` versus `mask[:, 1:]` target-mask alignment issue. Fixing the shift alone would not make the current full-sequence masks response-only.

## Possible fix

Preserve the response start position for both chosen and rejected sequences when constructing the dataset, then create masks that are:

- `0` over system/user prompt tokens;
- `1` over assistant response tokens;
- `0` over padding.
